src/components/ciConfig/BuildContext.tsx

```diff
@@ -0,0 +1,275 @@
+                trigger="click"
+                interactive
+                documentationLinkText="View Documentation"
+                additionalContent={getBuildContextAdditionalContent()}
+            >
+                <div className="icon-dim-16 fcn-5 ml-8 cursor">
+                    <Question />
+                </div>
+            </TippyCustomized>
+        </div>
+    )
+}
+
+export default function BuildContext({
+    readOnly,
+    isDefaultBuildContext,
+    configOverrideView,
+    sourceConfig,
+    selectedBuildContextGitMaterial,
+    currentMaterial,
+    setSelectedBuildContextGitMaterial,
+    repositoryError,
+    handleOnChangeConfig,
+    buildContextValue,
+    currentCIBuildConfig,
+    // Wanted to remove this prop but its getting updated
+    formState,
+    setCurrentCIBuildConfig,
+    currentBuildContextGitMaterial,
+    readOnlyBuildContextPath,
+}: BuildContextProps) {
+    const useRootBuildContextFlagFormState = currentCIBuildConfig?.useRootBuildContext
+    const [isCollapsed, setIsCollapsed] = useState<boolean>(!isDefaultBuildContext)
+    const [useRootBuildContextFlag, setUseRootBuildContextFlag] = useState<boolean>(useRootBuildContextFlagFormState)
+
+    const buildContextCheckoutPath = selectedBuildContextGitMaterial?.checkoutPath ?? currentMaterial?.checkoutPath
+    const checkoutPathArray = [{ label: RootBuildContext, value: RootBuildContext }]
+    if (buildContextCheckoutPath !== RootBuildContext) {
+        checkoutPathArray.push({ label: buildContextCheckoutPath, value: buildContextCheckoutPath })
+    }
+    const [checkoutPathOptions, setCheckoutPathOptions] = useState<OptionType[]>(checkoutPathArray)
+
+    useEffect(() => {
+        const checkoutPathArray = [{ label: RootBuildContext, value: RootBuildContext }]
+        if (selectedBuildContextGitMaterial?.checkoutPath !== RootBuildContext) {
+            checkoutPathArray.push({
+                label: selectedBuildContextGitMaterial?.checkoutPath,
+                value: selectedBuildContextGitMaterial?.checkoutPath,
+            })
+        }
+        setCheckoutPathOptions(checkoutPathArray)
+    }, [selectedBuildContextGitMaterial])
+
+    const toggleCollapse = () => {
+        setIsCollapsed(!isCollapsed)
+    }
+
+    const getSelectedBuildContextGitMaterial = () => selectedBuildContextGitMaterial ?? currentMaterial
+
+    const handleBuildContextPathChange = (selectedBuildContextGitMaterial): void => {
+        setSelectedBuildContextGitMaterial(selectedBuildContextGitMaterial)
+        // Don't know how and why we are directly setting state.
+        formState.repository.value = selectedBuildContextGitMaterial.name
+        setCurrentCIBuildConfig({
+            ...currentCIBuildConfig,
+            buildContextGitMaterialId: selectedBuildContextGitMaterial.id,
+        })
+    }
+
+    const handleBuildContextCheckoutPathChange = (checkoutPath) => {
+        const val = checkoutPath.value
+        let flag = false
```

Comment:
we can directly write it as `const flag = checkoutPath.value === RootBuildContext`

Reply:
Done